integrations/octopus/main.py

```diff
@@ -48,25 +40,57 @@ async def setup_application() -> None:
         )
         return
     octopus_client = await init_client()
-    existing_subscriptions = await octopus_client.get_webhook_subscriptions()
-    existing_webhook_uris = {
-        subscription.get("EventNotificationSubscription", {}).get("WebhookURI")
-        for subscription in existing_subscriptions
-    }
-    webhook_uri = f"{app_host}/integration/webhook"
-    if webhook_uri in existing_webhook_uris:
-        logger.info(f"Webhook already exists with URI: {webhook_uri}")
-    else:
-        await octopus_client.create_webhook_subscription(app_host)
-        logger.info(f"Webhook created with URI: {webhook_uri}")
+    async for spaces in octopus_client.get_all_spaces():
+        for space in spaces:
+            async for subscriptions in octopus_client.get_webhook_subscriptions(
+                space.get("Id")
```

Comment:
can we be defensive and catch possible None type of space.get('iD') before passing it to get_webhook_subscriptions? I want us to avoid scenario where we pass None to the API

Reply:
Done